k9mail/src/test/java/com/fsck/k9/activity/misc/ContactPictureLoaderTest.java

```diff
@@ -0,0 +1,59 @@
+package com.fsck.k9.activity.misc;
+
+
+import com.fsck.k9.K9RobolectricTestRunner;
+import com.fsck.k9.mail.Address;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(K9RobolectricTestRunner.class)
+public class ContactPictureLoaderTest {
+
+    @Test
+    public void calcUnknownContactLetter__matchesValidAscii() {
+        String result = ContactPictureLoader.calcUnknownContactLetter(new Address("a <a@b.com>"));
+        assertEquals("A", result);
+    }
+
+    @Test
+    public void calcUnknownContactLetter__matchesLstroke() {
+        String result = ContactPictureLoader.calcUnknownContactLetter(new Address("Ł <a@b.com>"));
+        assertEquals("Ł", result);
+    }
+
+
+    @Test
+    public void calcUnknownContactLetter__matchesChinese() {
+        String result = ContactPictureLoader.calcUnknownContactLetter(new Address("妈 <a@b.com>"));
+        assertEquals("妈", result);
+    }
+
+
+    @Test
+    public void calcUnknownContactLetter__matchesSurrogatePair() {
+        String result = ContactPictureLoader.calcUnknownContactLetter(new Address("\u0061\u0300 <a@b.com>"));
+        assertEquals("\u0041\u0300", result);
+    }
+
+    @Test
+    public void calcUnknownContactLetter__doesntMatchSpace() {
+        String result = ContactPictureLoader.calcUnknownContactLetter(new Address(" a <a@b.com>"));
+        assertEquals("A", result);
+    }
+
+
+    @Test
+    public void calcUnknownContactLetter__doesntMatchPunctuation() {
+        String result = ContactPictureLoader.calcUnknownContactLetter(new Address("-a <a@b.com>"));
+        assertEquals("A", result);
+    }
+
+    @Test
+    public void calcUnknownContactLetter__doesntMatchEmoji() {
+        String result = ContactPictureLoader.calcUnknownContactLetter(new Address("\uD83D\uDE00 <a@b.com>"));
```

Comment:
This is a surrogate pair. Just one that isn't a letter.

For a positive test you could use e.g. `"\uD800\uDFB5"`. It is a letter and renders fine with the default font on my desktop machine: 𐎵